.golangci-kal.yml

```diff
@@ -0,0 +1,56 @@
+version: "2"
+linters:
+  default: none
+  enable:
+    - kubeapilinter
+  settings:
+    custom:
+      kubeapilinter:
+        type: module
+        description: Kube API LInter lints Kube like APIs based on API conventions and best practices.
+        settings:
+          linters:
+            enable:
+            - "conditions" # Ensure conditions have the correct json tags and markers.
+            - "duplicatemarkers" # Ensure there are no exact duplicate markers. for types and fields.
+            - "integers" # Ensure only int32 and int64 are used for integers.
+            - "jsontags" # Ensure every field has a json tag.
+            - "maxlength" # Ensure all strings and arrays have maximum lengths/maximum items.
+            - "nobools" # Bools do not evolve over time, should use enums instead.
```

Comment:
@JoelSpeed would this preclude the flat boolean we're considering using for CORS [here](https://github.com/kubernetes-sigs/gateway-api/pull/3895)?

Reply:
Yes, it would. But of course, all of these are optional, and should you decide you don't want to exclude bools, you can disable this. The guidance upstream says bools should be considered and in most cases become enums, but there are some cases where bools are still allowed. In other projects they may be more strict, and this linter really is for the strict case.

Reply:
Looking over our `apis`/`apisx` it does seem that CORS is the only place we've used a bool. In that case there seems to be strong confidence however that the value will never be non-boolean. So I suppose what we would need to decide is whether we want to change that to a string enum.